packages/dcos-integration-test/extra/test_dcos_diagnostics.py

```diff
@@ -346,155 +361,122 @@ def _check_diagnostics_bundle_status(dcos_api_session):
-        masters=dcos_api_session.masters,
-        all_slaves=dcos_api_session.all_slaves,
-        session=dcos_api_session.copy().session,
-    )
-
     create_response = diagnostics.start_diagnostics_job().json()
     diagnostics.wait_for_diagnostics_job(last_datapoint=last_datapoint)
     diagnostics.wait_for_diagnostics_reports()
     bundles = diagnostics.get_diagnostics_reports()
-    assert len(bundles) == 1, 'bundle file not found'
-    assert bundles[0] == create_response['extra']['bundle_name']
+    assert len(bundles) > 0, 'bundle file not found'
 
-    return create_response['extra']['bundle_name']
+    bundle_name = create_response.get('id')
+    if not bundle_name:
+        bundle_name = create_response['extra']['bundle_name']
+    assert bundle_name in bundles
 
+    return bundle_name
 
-def _delete_bundle(dcos_api_session, bundle):
-    health_url = dcos_api_session.default_url.copy(
-        query='cache=0',
-        path='system/health/v1',
-    )
-    diagnostics = Diagnostics(
-        default_url=health_url,
-        masters=dcos_api_session.masters,
-        all_slaves=dcos_api_session.all_slaves,
-        session=dcos_api_session.copy().session,
-    )
 
+def _delete_bundle(diagnostics: Diagnostics, bundle):
     bundles = diagnostics.get_diagnostics_reports()
     assert bundle in bundles, 'not found {} in {}'.format(bundle, bundles)
 
-    dcos_api_session.health.post(os.path.join('/report/diagnostics/delete', bundle))
+    diagnostics.delete_bundle(bundle)
 
     bundles = diagnostics.get_diagnostics_reports()
     assert bundle not in bundles, 'found {} in {}'.format(bundle, bundles)
 
 
-@retrying.retry(wait_fixed=2000, stop_max_delay=LATENCY * 1000)
-def _download_and_extract_bundle(dcos_api_session, bundle):
-    _download_bundle_from_master(dcos_api_session, 0, bundle)
+def _download_and_extract_bundle(dcos_api_session, bundle, diagnostics):
+    _download_bundle_from_master(dcos_api_session, 0, bundle, diagnostics)
 
 
-@retrying.retry(wait_fixed=2000, stop_max_delay=LATENCY * 1000)
-def _download_and_extract_bundle_from_another_master(dcos_api_session, bundle):
+def _download_and_extract_bundle_from_another_master(dcos_api_session, bundle, diagnostics,):
     if len(dcos_api_session.masters) > 1:
-        _download_bundle_from_master(dcos_api_session, 1, bundle)
+        _download_bundle_from_master(dcos_api_session, 1, bundle, diagnostics)
 
 
-def _download_bundle_from_master(dcos_api_session, master_index, bundle):
+def _download_bundle_from_master(dcos_api_session, master_index, bundle, diagnostics):
     """ Download DC/OS diagnostics bundle from a master
 
     :param dcos_api_session: dcos_api_session fixture
     :param master_index: master index from dcos_api_session.masters array
     :param bundle: bundle name to download from master
+    :param diagnostics: DCOS Diagnostics client
     """
-    assert len(dcos_api_session.masters) >= master_index + 1, '{} masters required. Got {}'.format(
```

Comment:
Why is this assertion gone?

Reply:
Its not required the test will fail later when you try to read this value